invokeai/app/invocations/generate.py

```diff
@@ -57,19 +57,17 @@ def step_callback(sample, step=0):
         # Handle invalid model parameter
         # TODO: figure out if this can be done via a validator that uses the model_cache
         # TODO: How to get the default model name now?
-        if self.model is None or self.model == "":
-            self.model = context.services.generate.model_name
-
-        # Set the model (if already cached, this does nothing)
-        context.services.generate.set_model(self.model)
-
-        results = context.services.generate.prompt2image(
+        manager = context.services.model_manager
```

Comment:
I still feel like it would look nicer if you had a step to get the model, then did the generation using the model (as opposed to passing the manager around).

```py
model = context.services.model_manager.get_model(self.model)
generator = Txt2Img(model)
outputs = generate(
...
```

Reply:
Also, here you need to either switch to the model indicated by `self.model`, or you need to *get* that model and use it (like I suggested in the above comment). The model name is part of the node now - that way you can have graphs with different models and it'll switch correctly between models when running an entire graph.

Reply:
Thanks. I've fixed this. The paradigm is now:

```
model = manager.get_model('stable-diffusion-2.1')
txt2img = Text2Img(model)    # use default parameters
outputs = txt2img(generate(prompt='banana sushi')
```

This could be a single line, obviously:

`outputs = Txt2Img(manager.get_model('stable-diffusion-2.1')).generate(prompt='banana sushi')`